### PR TITLE
Make dual-list pf4 mapper elements customizable

### PR DESCRIPTION
**Pull request by fhlavac**
Added necessary props to make dual-list pf4 mapper elements customizable

@Hyperkid123

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@fhlavac I have renamed this to WIP. Once you have time, make all elements customizable like in other mappers. :bow:

**Review by Hyperkid123 — COMMENTED** [review]
Good job. Now you need to add these props to the typescript module definition and to the documentation.

The typings file is called `dual-list-select.d.ts` and all the definitions must reflect their source. So if target element for a props object is `div` it has to use react div typings. If its for a PF `Toolbar` it has to use PF ToolbarProps definitione etc. You can find example in other components/mappers.

The documentation md file is: `packages/react-renderer-demo/src/doc-components/examples-texts/pf4/dual-list.md`. It should be called `dual-list-select.md` so rename it please and add all the new props to it with corresponding links to original component docs if possible. You can look at the MUI variant for guidance.

**Review by rvsia — COMMENTED** [review]
As @Hyperkid123 d.ts field should be updated.

Also, you should update the documentation page for this component.

Otherwise, looks good!

File: packages/pf4-component-mapper/src/files/dual-list-select.js [hunk 1]
```diff
@@ -53,11 +57,11 @@ const List = ({ value, optionClick, noOptionsTitle, filterValue, filterValueText
           onClick={(e) => optionClick(e, value)}
           key={value}
           value={value}
+          {...ListItemProps}
+          {...ListItemPropsItem}
           className={`ddorg__pf4-component-mapper__dual-list-select-option ${
             selectedValues.includes(value) ? 'ddorg__pf4-component-mapper__dual-list-select-option-selected' : ''
```
Comment (rvsia) [inline packages/pf4-component-mapper/src/files/dual-list-select.js:63]: You should convert this into clsx function and add `ListItemProps.className` and `ListItemPropsItem.className`

File: packages/pf4-component-mapper/src/files/dual-list-select.d.ts [hunk 2]
```diff
@@ -25,15 +35,81 @@ interface InternalDualListSelectProps {
...
+  description?: ReactNode;
+  hideLabel?: boolean;
+  id?: string;
+  state?: AnyObject,
+  sortOptions?: VoidFunction,
+  filterOptions?: VoidFunction,
+  leftValues?: DualListSelectOption[];
+  rightValues?: DualListSelectOption[];
   renderStatus?: RenderStatusFunction;
+  handleOptionsClick?: HandleOptionsFunction,
```
Comment (Hyperkid123) [inline packages/pf4-component-mapper/src/files/dual-list-select.d.ts:56]: all of these handlers, sorter, etc, should not be part of the interface. These functions are inject from the common dual list select implementation imported from
```jsx
import DualListSelectCommon from '../../../common/src/dual-list-select';
```
and used at
```jsx
const DualListSelectWrapper = (props) => <DualListSelectCommon {...props} DualListSelect={DualList} />;
```
You can't rewrite those from the schema. Only props configurable from schema should be listed here.

I think any the stuff after the `// props` comment in the prop types definition.

File: packages/react-renderer-demo/src/doc-components/examples-texts/pf4/dual-list-select.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ Dual list select is wrapped in a form group, so it accepts all [form group props
 |Props|Type|Default|Description|
 |-----|----|-------|-----------|
 |options|array|[]|`[{label, value}]`|
+|label|node|FormLabel primary text|
```
Comment (Hyperkid123) [inline packages/react-renderer-demo/src/doc-components/examples-texts/pf4/dual-list-select.md:10]: I think you are missing one `|` between FormLabel and primary text.

**Review by Hyperkid123 — APPROVED** [review]

**Comment by rvsia** [comment]
Oh, there is no semantic-release commit message 😿

**Comment by Hyperkid123** [comment]
:facepalm: does not mapper. It will be released at some point. :D I don't think we need it atm.

**Pull request merged**